src/somnium/congomongo.clj

```diff
@@ -110,23 +110,38 @@
   "Makes a connection with passed database name, [{:host host, :port port}]
   server addresses and MongoClientOptions.
 
-  username and password may be supplied for authenticated connections."
-  [db {:keys [instances options username password]}]
+  username, password, and optionally auth-source, may be supplied for authenticated connections."
+  [db {:keys [instances options username password] {auth-mechanism :mechanism auth-source :source} :auth-source}]
```

Comment:
Nit, wondering if it might be clearer to rename `:source` -> `:db` or `:database` to keep closer in line w/ terminology in docs and [cli options](https://docs.mongodb.com/manual/reference/program/mongo/#authentication-options). You're more familiar w/ this than I am, but a quick read made me expect the term "database".

Reply:
I was going with java driver naming convention, e.g. 
`MongoCredential createScramSha256Credential(String userName, String source, char[] password)` and similarly when configuring connection via URI query param is `authSource`, e.g. `authSource=admin`